R/util-Report.R

```diff
@@ -6,15 +6,31 @@
   }
 }
 
-stop_if_empty <- function(x, x_arg = rlang::caller_arg(x)) {
-  if (!length(x)) {
-    LogMessage(
-      level = "fatal",
-      message = "{x_arg} must not be `NULL`."
-    )
+stop_if <- function(cnd, message) {
+  # Use enquo to capture the condition for lazy evaluation
```

Comment:
What's the point of `enquo` and `eval_tidy`-ing the logical expression?  Is there a scenario where a data mask is passed into `stop_if`?

Reply:
Think when I was initially drafting toy examples I needed to evaluate objects that may not necessarily have been initialized/don't exist yet, so it was necessary there, but now that I think about it, I suppose all the conditions we feed into cnd already are TRUE/FALSE. Let me see if this simplication applies and just spread it everywhere.

Reply:
Only other thought was maybe you want to pass the un-evaluated logical statement to the logger. Either way TIL about `enquo` and `eval_tidy` - cool stuff!